### PR TITLE
[CPU][DEBUG CAPS] Build fix

### PR DESCRIPTION
**Pull request by v-Golubev**
Build with enabled CPU debug capabilities is broken after PR [#15513](https://github.com/openvinotoolkit/openvino/pull/15513) merge into master. This PR reverts particular changes that introduce the build issue.

### CODE REVIEW

**Comment by v-Golubev** [comment]
@ilya-lavrenov FYI

File: src/plugins/intel_cpu/src/transformation_pipeline.h [hunk 1]
```diff
@@ -41,6 +43,7 @@ class Transformations {
     const bool    enableLpt;
     const bool    enableBF16;
     const bool    isLegacyApi;
+    const Config& config;
```
Comment (ilya-lavrenov) [inline src/plugins/intel_cpu/src/transformation_pipeline.h:46]: but as we can see - it's not used, is not it?

Comment (ilya-lavrenov) [inline src/plugins/intel_cpu/src/transformation_pipeline.h:46]: compiler reported this field is not used
looks like we need to make it happy differently.

have you tried to compile with clang?

**Review by EgorDuplensky — APPROVED** [review]

File: src/plugins/intel_cpu/src/transformation_pipeline.h (hunk 1 above)
Comment (EgorDuplensky) [reply src/plugins/intel_cpu/src/transformation_pipeline.h]: by the way, there are still warnings in the build, after -Wall was enabled.
Are there any plans to activation -Werror or something?

Comment (v-Golubev) [reply src/plugins/intel_cpu/src/transformation_pipeline.h]: @ilya-lavrenov thanks for the suggestion. I compiled the project without debug caps and fixed "unused field" warning.

**Review by ilya-lavrenov — APPROVED** [review]

File: src/plugins/intel_cpu/src/transformation_pipeline.h (hunk 1 above)
Comment (ilya-lavrenov) [reply src/plugins/intel_cpu/src/transformation_pipeline.h]: > by the way, there are still warnings in the build, after -Wall was enabled.
Are there any plans to activate -Werror or something?

We already use CMAKE_COMPILE_WARNING_AS_ERROR on CI

**Pull request merged**